h2o-genmodel-extensions/xgboost/src/main/java/hex/genmodel/algos/xgboost/XGBoostJavaMojoModel.java

```diff
@@ -156,13 +156,32 @@ public double getInitF() {
   }
 
   private final class XGBoostContributionsPredictor extends ContributionsPredictor<FVec> {
-    private XGBoostContributionsPredictor(TreeSHAPPredictor<FVec> treeSHAPPredictor) {
-      super(_nums + _catOffsets[_cats] + 1, treeSHAPPredictor);
+    private XGBoostContributionsPredictor(XGBoostMojoModel model, TreeSHAPPredictor<FVec> treeSHAPPredictor) {
+      super(_nums + _catOffsets[_cats] + 1, makeContributionNames(model), treeSHAPPredictor);
     }
 
     @Override
     protected FVec toInputRow(double[] input) {
       return _1hotFactory.fromArray(input);
     }
   }
+
+  private static String[] makeContributionNames(XGBoostMojoModel m) {
+    final String[] names = new String[m._nums + m._catOffsets[m._cats] + 1];
+    final String[] features = m.features();
+    int i = 0;
+    for (int c = 0; c < features.length; c++) {
+      if (m._domains[c] == null) {
+        names[i++] = features[c];
+      } else {
+        for (String d : m._domains[c])
+          names[i++] = features[c] + "." + d;
+        names[i++] = features[c] + ".missing(NA)";
```

Comment:
qq: Why is there this name for each categorical feature?

Reply:
this is because XGBoost uses 1-hot encoding, the output in H2O currently respects the one hot encoding

the naming schema respects the naming in h2o.predict_contribs

Reply:
ok, thanks for your explanation!